app/javascript/components/explore/GenomeView.js

```diff
@@ -0,0 +1,251 @@
+      let listToShow = bamFileList.bamAndBaiFiles
+      if (bamFileName) {
+        // if the user has specified a particular file name (likely because they are coming from the study download tab)
+        // then limit the lsit of files to show to just that one
+        listToShow = listToShow.filter(file => file.name === bamFileName)
+      }
+      const fileNamesToShow = listToShow.map(file => file.name).join(',')
+      // we only want to render igv when this tab is visible (igv can't draw itself to hidden panels)
+      // but we don't want to rerender every time the user toggles.
+      // So we track what the last files are that we initialized
+      // IGV with, and only rerender if they are different.
+      if (igvInitializedFiles !== fileNamesToShow) {
+        initializeIgv(igvContainerId, listToShow, bamFileList.gtfFiles)
+      }
+      setIgvInitializedFiles(fileNamesToShow)
+    }
+  }, [fileListString, bamFileName, isVisible])
+
+  /** handle clicks on the download 'browse in genome' buttons
+   * This should get refactored when/if we migrate the other study-overview tabs to react
+  */
+  useEffect(() => {
+    $(document).on('click', '.bam-browse-genome', e => {
+      $('#study-visualize-nav > a').click()
+      const selectedBam = $(e.target).attr('data-filename')
+      updateExploreParams({ bamFileName: selectedBam, tab: 'genome' })
+    })
+    $(document).on('click', '#study-visualize-nav > a', () => {
+      // IGV doesn't handle rendering to hidden divs. So for edge cases where this renders but is not shown
+      // (e.g. someone is viewing the genome tab, then navigates to summary tab, then reloads the page)
+      // we trigger a resize event so that IGV will know to redraw itself
+      if (isVisible) {
+        window.dispatchEvent(new Event('resize'))
+      }
+    })
+    return () => {
+      $(document).off('click', '.bam-browse-genome')
+    }
+  }, [])
+
+  /** show the full list of files, rather than the specific selected one */
+  function showAllFiles() {
+    updateExploreParams({ bamFileName: '' })
+  }
+
+  return <div>
+    { isLoading &&
+      <FontAwesomeIcon
+        icon={faDna}
+        data-testid="genome-view-loading-icon"
+        className="gene-load-spinner"
+      />
+    }
+    <div>
+      <div id={igvContainerId}></div>
+    </div>
+    { bamFileName && bamFileList?.length > 1 &&
+      <a className="action" onClick={showAllFiles}>See all sequence files for this study</a>
+    }
+  </div>
+}
+
+const SafeGenomeView = withErrorBoundary(GenomeView)
+export default SafeGenomeView
+
+
+/**
+ * Get tracks for selected BAM files, to show sequencing reads
+ */
+function getBamTracks(bamAndBaiFiles) {
+  let bam; let bamTrack; let i; let bamFileName
+
+  const bamTracks = []
+
+  for (i = 0; i < bamAndBaiFiles.length; i++) {
+    bam = bamAndBaiFiles[i]
+
+    // Extracts BAM file name from its GCS API URL
+    bamFileName = bam.url.split('/o/')[1].split('?')[0]
```

Comment:
Since we have control over what gets put in `bamAndBaiFiles` from `Study#get_bam_files`, why not add a field called `label` so we don't have to do this?

Reply:
hah, yup, I added the field, and then just didn't use it. Good catch